### ISSUE TITLE
DDO - API that returns in  SectorOnchainInfo 

### ISSUE THREAD

**Issue by jennijuju** [filecoin-project/lotus#11240]
DDO FIP proposes,"Adds a sector→deal mapping in the built-in market actor and deprecates the storage of deal IDs in the miner actor."

The new sector <> deal map in market actor goes as
```
/// HAMT[Address]HAMT[SectorNumber]SectorDealIDs
    pub provider_sectors: Cid,
```

**Impact 1: deal id is no longer stored in SectorOnchianInfo** 
**Impact 2: unsealed CIDs are added to sector unchain info**

APIs in lotus tha is impacted:
- StateMinerSectors
- StateMinerActiveSectors
- StateSectorGetInfo

I think we at least two options:
- 1️⃣ we could avoid breaking api changes, and get the deal ids of the given sectors from the new market actor `provider_sectors` map. given the map is indexed by miner address -> sector ID -> deal IDs, the computation complexity remains the same.
- 2️⃣ Lotus `State` APIs are designed to reflect whats stored in the state, and for `StateMiner__` methods, they reflects whats stored in miner actors. We could introduce a **breaking change** and have above methods no  longer returns deal info in sector info.  and users who needs this info may call https://github.com/filecoin-project/lotus/issues/11239.

**Comment by anorth** [comment]
I strongly recommend option 2. You won't be able to support option 1 forever, as subsequent changes happen. Rip off the band-aid.

**Comment by jennijuju** [comment]
Forgot to mention in my OP that my strong preference is also option 2. We are evolving sector/storage <> piece relationship and ideally ecosystem should adapt sooner rather than later. For lotus, from  a ref. Implementation perspective, I think it’s one of our product requirement to direct our user toward the right direction for the long term and support/ease ecosystem to adapt the new development. 


In addition, I also see biz/tooling that’s dedicated for f05-storage market ; or indexing/application/dashboard/marketplace that want to support listing all storage market. Imho lotus only needs to provide the API allowing users to get the right data, making the usage of that data efficient for their user could be a separate tooling ecosystem built on top of lotus.

**Comment by jennijuju** [comment]
> that collect metrics, especially around amount of data stored in Filecoin and "health" of a deal will be most affected. 


I don’t think so as we are not changing anything in the existing storage market. So if the targeted use case is just around f05 deals, things should just work as currently.
